### PR TITLE
use new procpool api 

### PR DESCRIPTION
**Pull request by theomonnom**

### CODE REVIEW

File: examples/voice-assistant/minimal_assistant.py [hunk 1]
```diff
@@ -29,10 +29,10 @@ async def entrypoint(ctx: JobContext):
...
-async def request_fnc(req: JobRequest) -> None:
+async def job_request_fnc(req: JobRequest) -> None:
     logging.info("received request %s", req)
-    await req.accept(entrypoint)
+    await req.accept()
 
 
 if __name__ == "__main__":
-    cli.run_app(WorkerOptions(request_fnc))
+    cli.run_app(WorkerOptions(job_request_fnc=job_request_fnc, job_entrypoint_fnc=entrypoint))
```
Comment (davidzhao) [inline examples/voice-assistant/minimal_assistant.py:38]: having `job_` prefix seems redundant? also, could it be left blank to auto accept?

```suggestion
    cli.run_app(WorkerOptions(entrypoint_fnc=entrypoint))
```

File: livekit-agents/livekit/agents/job.py [hunk 2]
```diff
@@ -44,8 +43,8 @@ class JobAcceptArguments:
 
 @dataclass
 class RunningJobInfo:
+    accept_arguments: JobAcceptArguments
```
Comment (davidzhao) [inline livekit-agents/livekit/agents/job.py:46]: what's the thinking in making these longer?

File: examples/voice-assistant/minimal_assistant.py (hunk 1 above)
Comment (theomonnom) [reply examples/voice-assistant/minimal_assistant.py:38]: we can discuss more about it tmrw

File: livekit-agents/livekit/agents/job.py (hunk 2 above)
Comment (theomonnom) [reply livekit-agents/livekit/agents/job.py:46]: this was public but not exposed to the users

**Pull request merged**